File: Examples/Ruby/Charts/update_chart_legend.rb
#ExStart:
require 'aspose_cells_cloud'

class Chart

  include AsposeCellsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @cells_api = CellsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end

  # Update chart legend
  def update_chart_legend
    file_name = "Sample_Test_Book.xls"
    upload_file(file_name)

    sheet_name = "Sheet5"
    chart_index = 0
    charts_legend = ChartsLegend.new
    font = Font.new
    color = Color.new
    color.a = "1"
    color.r = "255"
    color.g = "0"
    color.b = "0"
    font.color = color
    font.double_size = 10.0
    font.is_bold = true
    font.is_italic = false
    font.is_strikeout = false
    font.is_subscript = false
    font.is_superscript = false
    font.name = "Arial"
    font.size = 10
    font.size = 15
    font.underline = "None"
    charts_legend.font = font
    charts_legend.shadow = true        

    response = @cells_api.post_worksheet_chart_legend(file_name, sheet_name, chart_index, charts_legend)
  end

end

chart = Chart.new()
puts chart.update_chart_legend
#ExEnd:

File: Examples/Perl/Charts/UpdateChartLegend.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;
use AsposeCellsCloud::CellsApi;
use AsposeCellsCloud::ApiClient;
use AsposeCellsCloud::Configuration;
use AsposeCellsCloud::Object::Legend;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeCellsCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeCellsCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeCellsCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Cells API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $cellsApi = AsposeCellsCloud::CellsApi->new();

# Set input file name
my $name = 'Sample_Test_Book.xls';
my $sheetName = 'Sheet5';	
my $chartIndex = 0;
my @legendBody = AsposeCellsCloud::Object::Legend->new('Height' => 15, 'Position' => 'Left');

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);
# Invoke Aspose.Cells Cloud SDK API to update a chart legend of a chart                                          
$response = $cellsApi->PostWorksheetChartLegend(name=> $name, sheetName=>$sheetName, chartIndex=>$chartIndex, body=>@legendBody);

if($response->{'Status'} eq 'OK'){
	my $chartLegendInfo = $response->{'Legend'};
	print("\nChart Legend Position :: " . $chartLegendInfo->{'Position'});
}
#ExEnd:1